pkg/compare/secret_key_reference.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```

Comment:
nit: Rename this file. Maybe we could move it under `pkg/compare/secrets/reference.go`? Then when we do value comparison that can go under `value.go`?